src/parameters/modal_nucleation.jl

```diff
@@ -0,0 +1,85 @@
+"""
+    ModalNucleationParameters{FT}
+
+Parameters for the modal nucleation module
+
+# Fields
+$(DocStringExtensions.FIELDS)
+"""
+Base.@kwdef struct ModalNucleationParameters{FT} <: AbstractCloudMicrophysicsParameters
+    p_b_n::FT
+    p_b_i::FT
+    u_b_n::FT
+    u_b_i::FT
+    v_b_n::FT
+    v_b_i::FT
+    w_b_n::FT
+    w_b_i::FT
+    p_t_n::FT
+    p_t_i::FT
+    u_t_n::FT
+    u_t_i::FT
+    v_t_n::FT
+    v_t_i::FT
+    w_t_n::FT
+    w_t_i::FT
+    p_A_n::FT
+    p_A_i::FT
+    a_n::FT
+    a_i::FT
+    a_1::FT
+    a_2::FT
+    a_3::FT
+    a_4::FT
+    a_5::FT
+    Y_MTO3::FT
+    Y_MTOH::FT
+    k_MTO3::FT
+    k_MTOH::FT
+    exp_MTO3::FT
+    exp_MTOH::FT
+    k_H2SO4org::FT
+end
+
+ModalNucleationParameters(param_set) = 
+ModalNucleationParameters(
+    param_set.p_b_n,
+    param_set.p_b_i,
+    param_set.u_b_n,
+    param_set.u_b_i,
+    param_set.v_b_n,
+    param_set.v_b_i,
+    param_set.w_b_n,
+    param_set.w_b_i,
+    param_set.p_t_n,
+    param_set.p_t_i,
+    param_set.u_t_n,
+    param_set.u_t_i,
+    param_set.v_t_n,
+    param_set.v_t_i,
+    param_set.w_t_n,
+    param_set.w_t_i,
+    param_set.p_A_n,
+    param_set.p_A_i,
+    param_set.a_n,
+    param_set.a_i,
+    param_set.a_1,
+    param_set.a_2,
+    param_set.a_3,
+    param_set.a_4,
+    param_set.a_5,
+    param_set.Y_MTO3,
+    param_set.Y_MTOH,
+    param_set.k_MTO3,
+    param_set.k_MTOH,
+    param_set.exp_MTO3,
+    param_set.exp_MTOH,
+    param_set.k_H2SO4org,
+)
```

Comment:
```suggestion
function ModalNucleationParameters(param_set)
    fns = fieldnames(ModalNucleationParameters)
    vals = getproperty.(Ref(param_set), fns)
    nt = (;zip(fns, vals)...)
    ModalNucleationParameters(;nt...)
end
```
I think this should work (safely) too, if all of the names are consistently stored in `param_set`